### DISCUSSION TITLE
Open  mobile browser

### DISCUSSION THREAD

**Discussion by annunal** [beeware/toga#2352] [Q&A]
I am tryong to realize a method to check on a web site if there is a new version of my apk and in case download the file and...
After having downloaded the file,  I tried:
- use a toga.webview in which an hyperlink to the web site with the apk is present:  it refuses to open the file
- use a toga.webview containig an hyperlink to the donloaded file name:  it dows not work
- How can I open the mobile web browser from within the app ?

Any idea to achieve this ?
Thanks

-

**Comment by freakboy3742** [comment]
I'm unclear on what you're asking here.

If you want to open the web browser app on your device, you can use the `webbrowser` module on most operating systems. On Android, I think you'll need to do something like [this](https://stackoverflow.com/a/2201999), but using the native Chaquopy bridge.

If the question is how to detect if a new APK is available by checking a URL - there's no need to use a web browser at all. Use a HTTP library (my recommendation would be httpx, but requests can also work) to issue a request and inspect the result.

If you're expecting to be able to *install* an APK from a URL... good luck with that. What you're describing is an App Store, and they're not easy to build, because  of the massive security implications they have.

If you want your app to auto-update... upload it to an *actual* App Store.

**Reply by Tolsi73** [reply]
Hi, I'm facing the probelm you highlight: trying to use `webbrowser ` module to open the native webbroser on an Android device.

As I'm quite a noob, I'm not sure to understand what to do when you refer to [this](https://stackoverflow.com/questions/2201917/how-can-i-open-a-url-in-androids-web-browser-from-my-application/2201999#2201999) post.

How can I add the "Intent" using Chaquopy?

In advance, thank you

**Reply by freakboy3742** [reply]
The Intent API is documented [here](https://toga.readthedocs.io/en/stable/reference/platforms/android.html#activities-and-intents)

**Reply by Tolsi73** [reply]
Thanks !
